src/server/game/Anticheat/AnticheatMgr.h

```diff
@@ -0,0 +1,153 @@
+/*
+* Copyright (C) 2011-2016 TrinityCore <http://www.trinitycore.org/>
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SC_ACMGR_H
+#define SC_ACMGR_H
+
+#include "AnticheatData.h"
+#include "Common.h"
+#include "SharedDefines.h"
+#include "ScriptMgr.h"
+#include "Player.h"
+
+class ChatHandler;
+class AnticheatCheck;
+
+enum DetectionTypes
+{
+    SPEED_HACK_DETECTION            = 0x01,
+    FLY_HACK_DETECTION              = 0x02,
+    WALK_WATER_HACK_DETECTION       = 0x04,
+    JUMP_HACK_DETECTION             = 0x08,
+    TELEPORT_PLANE_HACK_DETECTION   = 0x10,
+    CLIMB_HACK_DETECTION            = 0x20
+};
+
+// GUIDLow is the key.
+typedef std::unordered_map<ObjectGuid::LowType, AnticheatData> AnticheatPlayersDataMap;
```

Comment:
Please do not use lowguid as key, it should only be used in database (I know there are MANY uses for it still in the code but don't add new ones)